src/main/java/com/es/phoneshop/model/product/DAO/ArrayListProductDao.java

```diff
@@ -0,0 +1,98 @@
+package com.es.phoneshop.model.product.DAO;
+
+import com.es.phoneshop.model.product.entity.Product;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Currency;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.stream.Collectors;
+
+public class ArrayListProductDao implements ProductDao {
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
+    private final Lock readLock = lock.readLock();
+    private final Lock writeLock = lock.writeLock();
+    private List<Product> products;
+
+    public ArrayListProductDao() {
```

Comment:
DAO should be a singleton
Otherwise if you have 10 servlets which require this DAO you will create 10 instances, which is not fine